src/env/node/git/localGitProvider.ts

```diff
@@ -4060,4 +4062,45 @@ export class LocalGitProvider implements GitProvider, Disposable {
 			`${prefix} requires a newer version of Git (>= ${version}) than is currently installed (${await this.git.version()}).${suffix}`,
 		);
 	}
+
+	@gate()
+	@log()
+	async getWorkDirStats(repoPath: string) {
+		const data = await this.git.ls_files(repoPath, {
+			deleted: true,
+			excludeStandard: true,
+			modified: true,
+			others: true,
+			showStatus: true,
+		});
+
+		if (data == undefined) return undefined;
+		const stats = { added: 0, deleted: 0, modified: 0 };
+		const statusByFile: Record<string, any> = {};
+		const statLines: string[] = data.split('\n');
+		for (const line of statLines) {
+			const [status, file] = line.split(' ');
+			let incrementCount = false;
+			if (statusByFile[file] === undefined) {
+				statusByFile[file] = status;
+				incrementCount = true;
+			} else if (statusByFile[file] === 'C' && status !== 'C') {
+				statusByFile[file] = status;
+				stats.modified = stats.modified - 1;
+				incrementCount = true;
+			}
```

Comment:
Note on this part: I noticed in git-ls-files that if a file is deleted, it shows up as both "modified" and "deleted". In this case we want to remove it from "modified" and count it as "deleted" instead.